File: Java & Python/linked_lists/reverse_linked_list_section.py
# find start pos, find prev, start reversing
# temp = prev, temp.next = prev, move to next nodes
class Solution:
    # func: reverse a linked list from position m to n
    # @param head: head node of the list
    # @param m: starting position m
    # @param n: ending position n
    # @return: new head node
    def reverseBetween(self, head, m, n):
        root = ListNode(0)
        root.next = head
        #Find position m
        prev = root
        count = 1
        while count < m:
            prev = prev.next
            count += 1
        start = prev
        end = start.next
     
        #Start reversing
        prev = prev.next
        temp = prev.next
        while count < n:
            next = temp.next
            temp.next = prev
            prev = temp
            temp = next
            count += 1
     
        start.next = prev
        end.next = curr
     
        return root.next

File: Java & Python/linked_lists/reverse_linked_list_section.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     public int val;
 *     public ListNode next;
 *     ListNode(int x) { val = x; next = null; }
 * }
 */
public class Solution {
	public ListNode reverseBetween(ListNode A, int m, int n) {
		int i = 1;
		ListNode root = new ListNode(0);
		ListNode prev, next, current, start, end;

		root.next = A;
		prev = root;
		
		while (i < m) {
			prev = prev.next;
			++i;
		}
		start = prev;
		end = prev.next;

		// start reversing
		prev = prev.next;
		current = prev.next;
		while (i < n) {
			next = current.next;
			current.next = prev;
			prev = current;
			current = next;
			++i;
		}
		start.next = prev;
		end.next = current;
		return root.next;
	}
}